File: p1/c_version/p1.c
/*Project Description:
 * If we list all the natural numbers below 10 that are multiples of 3 or 5,
 * we get 3, 5, 6 and 9. The sum of these multiples is 23.
 *  
 * Find the sum of all the multiples of 3 or 5 below 1000.
 * */

int main(){
	int result = sum(1000);
	printf("The sum of all the multiples of %d or %d below %d is: %d\n", 3, 5, 1000, result);

}

int sum(int n){
	//Creat for loop the checks which digits are multiples of the number being
	//counted, then added the sum up per loop
	int count = 0;
	for(int i = 0; i < n; i++){
		if(i % 3 == 0 || i % 5 == 0){
			count = count + i;
		}

	}
	return count;

}

File: p1/rust_version/p1.rs
/*Project Description:
 * If we list all the natural numbers below 10 that are multiples of 3 or 5,
 * we get 3, 5, 6 and 9. The sum of these multiples is 23.
 *  
 * Find the sum of all the multiples of 3 or 5 below 1000.
 * */

 fn main(){
 	let answer = sum(1000);
 	println!("The sum of all multiples of 3 or 5 below 1000 is: {answer}");



 }
//Answer should be 233168
 fn sum(n: i32) -> i32{

 	//The created for loop checks which digits are multiples of the number being
	//counted, then added the sum up per loop
	let mut i = 0;
	let mut count = 0;
	while i < n {
		if i % 3 == 0 || i % 5 == 0 {
			count = count + i;

		}
		i+=1;


	}
	return count;

 }